### DISCUSSION TITLE
[Unknown Mod][Castles] Enhanced information with additional thieves' guilds

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#8850] [Ideas]
In Heroes 2, there are five thieves' guilds to build in order to obtain all possible information.
After this number, there's no point in building more thieves' guilds to get more information.
This change increases the number of thieves' guilds of interest to twelve, giving additional quantity information on the various categories.
Images are for explanation only.

In fheroes2:

<details>
<summary> In fheroes2 (current screens): (click to expand)</summary>
<br>

With one thieves' guild:
![Fheroes2 thieve guild info 001](https://github.com/ihhub/fheroes2/assets/43583994/77859696-9fb0-41fc-9f5c-342412c72a6a)

With two thieves' guilds:
![Fheroes2 thieve guild info 002](https://github.com/ihhub/fheroes2/assets/43583994/a189e912-5b1f-43b5-8ebf-8287be59c988)

With three thieves' guilds:
![Fheroes2 thieve guild info 003](https://github.com/ihhub/fheroes2/assets/43583994/d3186adf-c3a8-4fe5-9db2-3475436cd97d)

With four thieves' guilds:
![Fheroes2 thieve guild info 004](https://github.com/ihhub/fheroes2/assets/43583994/23853c0c-b70c-40fc-b91a-c4dfea55db4f)

With five thieves' guilds:
![Fheroes2 thieve guild info 005](https://github.com/ihhub/fheroes2/assets/43583994/cab428e7-cf06-4e8f-93d8-721ccd71f5e8)

</details>
<details>
<summary> Additional screens: (click to expand)</summary>
<br>

With six thieves' guilds:
![Fheroes2 thieve guild info 005 006](https://github.com/ihhub/fheroes2/assets/43583994/1aa82a83-0112-4c2b-9e20-7c1d6c2ee645)

With seven thieves' guilds:
![Fheroes2 thieve guild info 005 007](https://github.com/ihhub/fheroes2/assets/43583994/fea682ad-ae7c-49fe-8237-4ae7df41b543)

With eight thieves' guilds:
![Fheroes2 thieve guild info 005 008](https://github.com/ihhub/fheroes2/assets/43583994/92473cdc-9473-48eb-9155-9979dd373272)

With nine thieves' guilds:
![Fheroes2 thieve guild info 005 009](https://github.com/ihhub/fheroes2/assets/43583994/cf9fd9a4-d5b7-437e-b207-319e3f6b7690)

With ten thieves' guilds:
For creatures, the number represents all the creatures of the same type recruited.
![Fheroes2 thieve guild info 005 010](https://github.com/ihhub/fheroes2/assets/43583994/fb7fd492-f727-467d-83e3-e9e264386983)

With eleven thieves' guilds:
Left and right arrows are added near the hero portrait to cycle through each player's heroes.
These arrows are displayed only if the player has at least two heroes.
Note: these arrows are added to the player's hero portrait as of the first thieves' guild.
A number next to the hero portrait at the top right indicates the position in the hierarchy (first best hero, second best hero, etc.).
These numbers are displayed only if the player has at least two heroes.
<img width="656" height="518" alt="Fheroes2 thieve guild info 005 011" src="https://github.com/user-attachments/assets/0c8a4c6d-dc29-43fb-9fb3-7fb130ab7732" />

With twelve thieves' guilds:
Left and right arrows are added near the creature sprite to cycle through each player's creatures.
These arrows are displayed only if the player has at least two types of creatures.
Note: these arrows are added to the player's creature sprite as of the first thieves' guild.
A number next to the creature sprite at the top right indicates the position in the hierarchy (first best creature, second best creature, etc.).
These numbers are displayed only if the player has at least two types of creatures.
<img width="656" height="518" alt="Fheroes2 thieve guild info 005 012" src="https://github.com/user-attachments/assets/b3cd23dd-9979-45a9-9829-06e81203a253" />

</details>

This is not optional.
This can also be done by default in fheroes2.

**Comment by pazam-h2** [comment]
A bit of a niche situation (how many times did you ever had more than 7 thieves' guilds?) for the anount of work it takes (several UI/UX implmentations).

Overall a great idea, and a good looking mockup, just probably a low priority idea. 

I suggest maybe in a future balance mod to the engine- this could be combined with scouting secondary skill in some way for better scouting + thieves' guild, without the need of 7+ thieves' guilds.

**Reply by LeHerosInconnu** [reply]
Hello @pazam-h2,

The aim is to encourage and reward players to build more thieves' guilds.
In the original game, once the number of useful guilds is reached, there is no longer any point in building them.

On larger map sizes, it's often possible to own many castles.

This will eventually be part of a mod for fheroes2 that will be released when modding becomes available.
See here: https://github.com/ihhub/fheroes2/discussions/2516.
